diff --git a/arch/arm/mach-omap2/board-ti8148evm.c b/arch/arm/mach-omap2/board-ti8148evm.c
index 94b4f38..8793041 100644
--- a/arch/arm/mach-omap2/board-ti8148evm.c
+++ b/arch/arm/mach-omap2/board-ti8148evm.c
@@ -530,112 +530,62 @@ static struct snd_platform_data ti8148_evm_snd_data = {
 };
 
 /* NOR Flash partitions */
-static struct mtd_partition ti814x_evm_norflash_partitions[] = {
-	/* bootloader (U-Boot, etc) in first 5 sectors */
-	{
-		.name		= "bootloader",
-		.offset		= 0,
-		.size		= 2 * SZ_128K,
-		.mask_flags	= MTD_WRITEABLE, /* force read-only */
-	},
-	/* bootloader params in the next 1 sectors */
-	{
-		.name		= "env",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= SZ_128K,
-		.mask_flags	= 0,
-	},
-	/* kernel */
-	{
-		.name		= "kernel",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= 2 * SZ_2M,
-		.mask_flags	= 0
-	},
-	/* file system */
-	{
-		.name		= "filesystem",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= 25 * SZ_2M,
-		.mask_flags	= 0
-	},
-	/* reserved */
-	{
-		.name		= "reserved",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= MTDPART_SIZ_FULL,
-		.mask_flags	= 0
-	}
-};
+static struct mtd_partition ti814x_evm_norflash_partitions[] = {};
 
 /* NAND flash information */
 static struct mtd_partition ti814x_nand_partitions[] = {
-/* All the partition sizes are listed in terms of NAND block size */
-	{
-		.name           = "U-Boot-min",
-		.offset         = 0,    /* Offset = 0x0 */
-		.size           = SZ_128K,
-		.mask_flags     = MTD_WRITEABLE,        /* force read-only */
-	},
-	{
-		.name           = "U-Boot",
-		.offset         = MTDPART_OFS_APPEND,/* Offset = 0x0 + 128K */
-		.size           = 18 * SZ_128K,
-		.mask_flags     = MTD_WRITEABLE,        /* force read-only */
-	},
-	{
-		.name           = "U-Boot Env",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x260000 */
-		.size           = 1 * SZ_128K,
-	},
-	{
-		.name           = "Kernel",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x280000 */
-		.size           = 34 * SZ_128K,
-	},
-	{
-		.name           = "File System",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x6C0000 */
-		.size           = 1601 * SZ_128K,
-	},
-	{
-		.name           = "Reserved",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0xCEE0000 */
-		.size           = MTDPART_SIZ_FULL,
-	},
+    /* bootloader       -       0x0 */
+    {
+        .name		= "bootloader",
+        .offset		= 0,
+        .size		= 1 * SZ_1M,
+    },
+    /* bootconfig       -   0x100000 */
+    {
+        .name		= "bootconfig",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 1 * SZ_1M,
+    },
+    /* first kernel     -   0x200000 */
+    {
+        .name		= "kernel1",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 10 * SZ_1M,
+        .mask_flags	= 0,
+    },
+    /* second kernel    -   0xc00000 */
+    {
+        .name		= "kernel2",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 10 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* root file system -   0x1600000 */
+    {
+        .name		= "rootfs1",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 100 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* root file system -   0x8000000 */
+    {
+        .name		= "rootfs2",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 100 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* appfs;           -   0xde00000
+     * => 35 MBytes w/ 255 MByte nand; 291 MBytes w/ 512 MByte NAND */
+    {
+        .name		= "application_fs",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= MTDPART_SIZ_FULL,
+        .mask_flags	= 0
+    }
 };
 
 /* SPI fLash information */
-struct mtd_partition ti8148_spi_partitions[] = {
-	/* All the partition sizes are listed in terms of erase size */
-	{
-		.name		= "U-Boot-min",
-		.offset		= 0,	/* Offset = 0x0 */
-		.size		= 32 * SZ_4K,
-		.mask_flags	= MTD_WRITEABLE, /* force read-only */
-	},
-	{
-		.name		= "U-Boot",
-		.offset		= MTDPART_OFS_APPEND, /* 0x0 + (32*4K) */
-		.size		= 64 * SZ_4K,
-		.mask_flags	= MTD_WRITEABLE, /* force read-only */
-	},
-	{
-		.name		= "U-Boot Env",
-		.offset		= MTDPART_OFS_APPEND, /* 0x40000 + (32*4K) */
-		.size		= 2 * SZ_4K,
-	},
-	{
-		.name		= "Kernel",
-		.offset		= MTDPART_OFS_APPEND, /* 0x42000 + (32*4K) */
-		.size		= 640 * SZ_4K,
-	},
-	{
-		.name		= "File System",
-		.offset		= MTDPART_OFS_APPEND, /* 0x2C2000 + (32*4K) */
-		.size		= MTDPART_SIZ_FULL, /* size ~= 1.1 MiB */
-	}
-};
+struct mtd_partition ti8148_spi_partitions[] = {};
 
 const struct flash_platform_data ti8148_spi_flash = {
 	.type		= "w25x32",
@@ -776,15 +726,17 @@ static void __init ti8148_evm_init(void)
 	/* initialize usb */
 	usb_musb_init(&musb_board_data);
 
-	ti8148_spi_init();
+	/* DNS: disable SPI. No SPI NAND on HidaV 
+	ti8148_spi_init(); */
 	ti814x_vpss_init();
 	ti814x_hdmi_init();
 #ifdef CONFIG_SND_SOC_TI81XX_HDMI
 	platform_add_devices(ti8148_devices, ARRAY_SIZE(ti8148_devices));
 #endif
 	regulator_use_dummy_regulator();
+	/* DNS: disable NOR. No NOR flash on HidaV
 	board_nor_init(ti814x_evm_norflash_partitions,
-		ARRAY_SIZE(ti814x_evm_norflash_partitions), 0);
+		ARRAY_SIZE(ti814x_evm_norflash_partitions), 0); */
 }
 
 static void __init ti8148_evm_map_io(void)
