diff --git a/arch/arm/mach-omap2/board-ti8168evm.c b/arch/arm/mach-omap2/board-ti8168evm.c
index ff6b74a..2ffb529 100644
--- a/arch/arm/mach-omap2/board-ti8168evm.c
+++ b/arch/arm/mach-omap2/board-ti8168evm.c
@@ -73,43 +73,7 @@ static struct omap2_hsmmc_info mmc[] = {
 #define VPS_VC_IO_EXP_THS7368_FILTER2_MASK  (0x80u)
 #define VPS_VC_IO_EXP_THS7368_FILTER_SHIFT  (0x06u)
 
-static struct mtd_partition ti816x_evm_norflash_partitions[] = {
-	/* bootloader (U-Boot, etc) in first 5 sectors */
-	{
-		.name		= "bootloader",
-		.offset		= 0,
-		.size		= 2 * SZ_128K,
-		.mask_flags	= MTD_WRITEABLE, /* force read-only */
-	},
-	/* bootloader params in the next 1 sectors */
-	{
-		.name		= "env",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= SZ_128K,
-		.mask_flags	= 0,
-	},
-	/* kernel */
-	{
-		.name		= "kernel",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= 2 * SZ_2M,
-		.mask_flags	= 0
-	},
-	/* file system */
-	{
-		.name		= "filesystem",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= 25 * SZ_2M,
-		.mask_flags	= 0
-	},
-	/* reserved */
-	{
-		.name		= "reserved",
-		.offset		= MTDPART_OFS_APPEND,
-		.size		= MTDPART_SIZ_FULL,
-		.mask_flags	= 0
-	}
-};
+static struct mtd_partition ti816x_evm_norflash_partitions[] = { };
 
 
 /* Maximum ports supported by the PCF8575 */
@@ -147,32 +111,55 @@ static struct mtd_partition ti816x_evm_norflash_partitions[] = {
 #define VR_GPIO_INSTANCE	0
 
 static struct mtd_partition ti816x_nand_partitions[] = {
-/* All the partition sizes are listed in terms of NAND block size */
-	{
-		.name           = "U-Boot",
-		.offset         = 0,    /* Offset = 0x0 */
-		.size           = 19 * NAND_BLOCK_SIZE,
-	},
-	{
-		.name           = "U-Boot Env",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x260000 */
-		.size           = 1 * NAND_BLOCK_SIZE,
-	},
-	{
-		.name           = "Kernel",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x280000 */
-		.size           = 34 * NAND_BLOCK_SIZE,
-	},
-	{
-		.name           = "File System",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0x6C0000 */
-		.size           = 1601 * NAND_BLOCK_SIZE,
-	},
-	{
-		.name           = "Reserved",
-		.offset         = MTDPART_OFS_APPEND,   /* Offset = 0xCEE0000 */
-		.size           = MTDPART_SIZ_FULL,
-	},
+    /* bootloader       -       0x0 */
+    {
+        .name		= "bootloader",
+        .offset		= 0,
+        .size		= 1 * SZ_1M,
+    },
+    /* first kernel     -   0x100000 */
+    {
+        .name		= "bootinfo",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 1 * SZ_1M,
+        .mask_flags	= 0,
+    },
+    /* first kernel     -   0x200000 */
+    {
+        .name		= "kernel1",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 10 * SZ_1M,
+        .mask_flags	= 0,
+    },
+    /* second kernel    -   0xc00000 */
+    {
+        .name		= "kernel2",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 10 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* root file system -   0x1600000 */
+    {
+        .name		= "rootfs1",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 100 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* root file system -   0x8000000 */
+    {
+        .name		= "rootfs2",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= 100 * SZ_1M,
+        .mask_flags	= 0
+    },
+    /* appfs;           -   0xde00000
+     * => 35 MBytes w/ 256 MByte nand; 291 MBytes w/ 512 MByte NAND */
+    {
+        .name		= "application_fs",
+        .offset		= MTDPART_OFS_APPEND,
+        .size		= MTDPART_SIZ_FULL,
+        .mask_flags	= 0
+    }
 };
 
 static struct at24_platform_data eeprom_info = {
@@ -654,30 +641,7 @@ static int __init ti816x_evm_i2c_init(void)
 }
 
 /* SPI fLash information */
-struct mtd_partition ti816x_spi_partitions[] = {
-	/* All the partition sizes are listed in terms of NAND block size */
-	{
-		.name		= "U-Boot",
-		.offset		= 0,	/* Offset = 0x0 */
-		.size		= 64 * SZ_4K,
-		.mask_flags	= MTD_WRITEABLE,	/* force read-only */
-	},
-	{
-		.name		= "U-Boot Env",
-		.offset		= MTDPART_OFS_APPEND,	/* Offset = 0x40000 */
-		.size		= 2 * SZ_4K,
-	},
-	{
-		.name		= "Kernel",
-		.offset		= MTDPART_OFS_APPEND,	/* Offset = 0x42000 */
-		.size		= 640 * SZ_4K,
-	},
-	{
-		.name		= "File System",
-		.offset		= MTDPART_OFS_APPEND,	/* Offset = 0x2C2000 */
-		.size		= MTDPART_SIZ_FULL,	/* size = 1.24 MiB */
-	}
-};
+struct mtd_partition ti816x_spi_partitions[] = { };
 
 const struct flash_platform_data ti816x_spi_flash = {
 	.name		= "spi_flash",
@@ -848,7 +812,6 @@ static void __init ti8168_evm_init(void)
 	ti816x_evm_i2c_init();
 	i2c_add_driver(&ti816xevm_cpld_driver);
 	ti81xx_register_mcasp(0, &ti8168_evm_snd_data);
-	ti816x_spi_init();
 	/* initialize usb */
 	usb_musb_init(&musb_board_data);
 
@@ -867,8 +830,6 @@ static void __init ti8168_evm_init(void)
 		ARRAY_SIZE(ti816x_nand_partitions), 0, NAND_BUSWIDTH_16);
 
 	omap2_hsmmc_init(mmc);
-	board_nor_init(ti816x_evm_norflash_partitions,
-		ARRAY_SIZE(ti816x_evm_norflash_partitions), 0);
 	ti816x_vpss_init();
 	ti816x_gpio_vr_init();
 	ti816x_hdmi_init();
