diff --git a/drivers/mtd/Kconfig b/drivers/mtd/Kconfig
index 1e2cbf5..bcfd8fc 100644
--- a/drivers/mtd/Kconfig
+++ b/drivers/mtd/Kconfig
@@ -226,6 +226,42 @@ config MTD_BLOCK_RO
 	  You do not need this option for use with the DiskOnChip devices. For
 	  those, enable NFTL support (CONFIG_NFTL) instead.
 
+config MTD_BLOCK_ROM_BBFREE
+	tristate "Bad-block-skipiping readonly 'blockrom' access to MTD devices"
+	depends on MTD_BLOCK!=y && MTD
+	help
+	  The blockrom MTD driver provides bad block free read access to MTDs via
+	  /dev/blockromX device files. Bad blocks are automatically skipped upon read,
+	  and reading continues with the next block.
+
+	  This allows for read-only filesystems to exist in MTDs with bad blocks:
+	  usually the baddies are skipped when writing the filesystem into MTD by
+	  the tool performing the write (e.g. nandwrite). To successfully mount 
+	  these filesystems, however, you will need a translation layer that skips 
+	  bad blocks upon read as well. blockrom is such a translation layer.
+
+config MTD_BLOCK_ROM_DELAYED_INIT
+	bool "Blockrom delayed initialisation"
+	depends on MTD_BLOCK_ROM_BBFREE
+	help
+	  This option will cause the bad-block-skipping readonly blockrom MTD
+	  translation layer to do its initial bad block scan asynchronously in a 
+	  background kernel thread instead of synchronously upon initialisation.
+
+	  Selecting this option may speed up boot time since lower index MTDs 
+	  are already usable (e.g. for root fs mount) while higher index MTDs 
+	  are still being scanned.
+
+config MTD_BLOCK_ROM_BIT_SCRUBBING
+	bool "Blockrom bit scrubbing support"
+	depends on MTD_BLOCK_ROM_BBFREE
+	help
+	  (NOTE: This feature is not yet implemented)
+	  This option will add bit scrubbing support to the bad-block-skipping 
+	  readonly blockrom MTD translation layer. The bit scrubber will erase
+	  and re-write an erase block if a corrected bit error was encountered
+	  upon read.
+
 config FTL
 	tristate "FTL (Flash Translation Layer) support"
 	depends on BLOCK
diff --git a/drivers/mtd/Makefile b/drivers/mtd/Makefile
index 760abc5..2337097 100644
--- a/drivers/mtd/Makefile
+++ b/drivers/mtd/Makefile
@@ -19,6 +19,7 @@ obj-$(CONFIG_MTD_CHAR)		+= mtdchar.o
 obj-$(CONFIG_MTD_BLKDEVS)	+= mtd_blkdevs.o
 obj-$(CONFIG_MTD_BLOCK)		+= mtdblock.o
 obj-$(CONFIG_MTD_BLOCK_RO)	+= mtdblock_ro.o
+obj-$(CONFIG_MTD_BLOCK_ROM_BBFREE)	+= blockrom.o
 obj-$(CONFIG_FTL)		+= ftl.o
 obj-$(CONFIG_NFTL)		+= nftl.o
 obj-$(CONFIG_INFTL)		+= inftl.o
