diff --git a/arch/arm/include/asm/arch-ti81xx/mem.h b/arch/arm/include/asm/arch-ti81xx/mem.h
index b4aa308..188564b 100644
--- a/arch/arm/include/asm/arch-ti81xx/mem.h
+++ b/arch/arm/include/asm/arch-ti81xx/mem.h
@@ -95,6 +95,7 @@ enum {
 #define SMNAND_GPMC_CONFIG6	0x1F0F0A80
 #define SMNAND_GPMC_CONFIG7	0x00000C44
 
+/*
 #define M_NAND_GPMC_CONFIG1	0x00001810
 #define M_NAND_GPMC_CONFIG2 	0x001e1e00
 #define M_NAND_GPMC_CONFIG3 	0x001e1e00
@@ -102,6 +103,15 @@ enum {
 #define M_NAND_GPMC_CONFIG5 	0x00151e1e
 #define M_NAND_GPMC_CONFIG6	0x16000f80
 #define M_NAND_GPMC_CONFIG7	0x00000008
+*/
+
+#define M_NAND_GPMC_CONFIG1 0x00001810
+#define M_NAND_GPMC_CONFIG2 0x00020200
+#define M_NAND_GPMC_CONFIG3 0x00020200
+#define M_NAND_GPMC_CONFIG4 0x01000100
+#define M_NAND_GPMC_CONFIG5 0x00020202
+#define M_NAND_GPMC_CONFIG6 0x00000000
+#define M_NAND_GPMC_CONFIG7 0x00000008
 
 #define STNOR_GPMC_CONFIG1	0x3
 #define STNOR_GPMC_CONFIG2	0x00151501
