diff --git a/lib/Makefile b/lib/Makefile
index c45f07c..be518e6 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -32,6 +32,9 @@ COBJS-$(CONFIG_BZIP2) += bzlib_decompress.o
 COBJS-$(CONFIG_BZIP2) += bzlib_randtable.o
 COBJS-$(CONFIG_BZIP2) += bzlib_huffman.o
 COBJS-$(CONFIG_USB_TTY) += circbuf.o
+COBJS-y += boot_config.o
+COBJS-y += lowlevel.o
+COBJS-y += logging.o
 COBJS-y += crc16.o
 COBJS-y += crc32.o
 COBJS-y += ctype.o
