
diff --git a/common/main.c b/common/main.c
index cd942d7..abdb0ab 100644
--- a/common/main.c
+++ b/common/main.c
@@ -381,39 +381,59 @@ void main_loop (void)
 	int ret = NULL;
 	enum bt_ll_parttype which=kernel;  /* { kernel, rootfs }; */
 
-	bc_ll_init( &bc , "/dev/dummy" );
-        btblock_test = bc_ll_get_current ( &bc, &block_index );
-
-	printf ("NAND-BootConfig:\n");
-        printf ("Epoch = %d\n", btblock_test->epoch);
-        printf ("Kernel booted = %d\n", btblock_test->kernel.n_booted );
-        printf ("Kernel healthy = %d\n", btblock_test->kernel.n_healthy );
-        printf ("Rootfs booted = %d\n", btblock_test->rootfs.n_booted );
-        printf ("Rootfs healthy = %d\n", btblock_test->rootfs.n_healthy );
-
-        /* rootfs_device=/dev/blockrom4\0       */
-	/* kernel_offset=0x200000\0 */
-        if( btblock_test->kernel.partition == 1 )
-	{
-		setenv ("kernel_offset", "0xC00000\0");
-		printf ("kernel_offset=0xC00000\n");
-	}
-	else
-	{
-		setenv ("kernel_offset", "0x200000\0");
-		printf ("kernel_offset=0x200000\n");
-	}
+	bc_ll_init( &bc , "/dev/mtd1" );
+	btblock_test = bc_ll_get_current ( &bc, &block_index );
 
-	if( btblock_test->rootfs.partition == 1 )
+	if ( btblock_test != 0)
 	{
-                setenv ("rootfs_device", "/dev/blockrom5\0");
-		printf ("rootfs_device=\"/dev/blockrom5\"\n");
-        }
-        else
-        {
-		setenv ("rootfs_device", "/dev/blockrom4\0");
-		printf ("rootfs_device=\"/dev/blockrom4\"\n");
-        }
+		printf ("NAND-BootConfig:\n");
+	        printf ("Epoch = %d\n", btblock_test->epoch);
+	        printf ("Kernel booted = %d\n", btblock_test->kernel.n_booted );
+	        printf ("Kernel healthy = %d\n", btblock_test->kernel.n_healthy );
+	        printf ("Rootfs booted = %d\n", btblock_test->rootfs.n_booted );
+	        printf ("Rootfs healthy = %d\n", btblock_test->rootfs.n_healthy );
+
+	        /* rootfs_device=/dev/blockrom4\0       */
+		/* kernel_offset=0x200000\0 */
+	        if( btblock_test->kernel.partition == 1 )
+		{
+			setenv ("kernel_offset", "0xC00000\0");
+			printf ("kernel_offset=0xC00000\n");
+		}
+		else
+		{
+			setenv ("kernel_offset", "0x200000\0");
+			printf ("kernel_offset=0x200000\n");
+		}
+
+		if( btblock_test->rootfs.partition == 1 )
+		{
+	                setenv ("rootfs_device", "/dev/blockrom5\0");
+			printf ("rootfs_device=\"/dev/blockrom5\"\n");
+	        }
+	        else
+	        {
+			setenv ("rootfs_device", "/dev/blockrom4\0");
+			printf ("rootfs_device=\"/dev/blockrom4\"\n");
+	        }
+
+	        if ( (btblock_test->kernel.n_booted == 1) || (btblock_test->rootfs.n_booted == 1) )
+	        {
+			printf ("First Boot of Epoch %d configuration.\n", btblock_test->epoch);
+			if (btblock_test->kernel.n_booted == 1)
+				printf ("Set Kernel-booted-Flag of Epoch %d configuration.\n", btblock_test->epoch);
+			if (btblock_test->rootfs.n_booted == 1)
+				printf ("Set Rootfs-booted-Flag of Epoch %d configuration.\n", btblock_test->epoch);
+
+	                bc_ll_set_booted_flag_on_current( &bc );
+	        	printf ("NEW NAND-BootConfig:\n");
+		        printf ("Epoch = %d\n", btblock_test->epoch);
+	        	printf ("Kernel booted = %d\n", btblock_test->kernel.n_booted );
+	        	printf ("Kernel healthy = %d\n", btblock_test->kernel.n_healthy );
+	        	printf ("Rootfs booted = %d\n", btblock_test->rootfs.n_booted );
+	        	printf ("Rootfs healthy = %d\n", btblock_test->rootfs.n_healthy );
+	        }
+	}
 
 }
 #endif
diff --git a/include/configs/ti8148_evm.h b/include/configs/ti8148_evm.h
index a4539da..fb3c127 100644
--- a/include/configs/ti8148_evm.h
+++ b/include/configs/ti8148_evm.h
@@ -23,7 +23,7 @@
 #define CONFIG_TI814X_EVM_DDR3
 
 #define CONFIG_ENV_SIZE			0x800
-# define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + (8 * 1024))
+# define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + (196 * 1024))
 #define CONFIG_SYS_GBL_DATA_SIZE	128	/* size in bytes reserved for initial data */
 
 #ifndef TI81XX_SECOND_STAGE

