diff --git a/drivers/mtd/nand/nand_base.c b/drivers/mtd/nand/nand_base.c
index 67190d5..aeb7cfd 100644
--- a/drivers/mtd/nand/nand_base.c
+++ b/drivers/mtd/nand/nand_base.c
@@ -478,16 +478,24 @@ static int nand_block_checkbad(struct mtd_info *mtd, loff_t ofs, int getchip,
 {
 	struct nand_chip *chip = mtd->priv;
 
+    /* 
+     * tfm: bad block table creation fails for BCH8 
+     * because generic nand_do_read_oob() does not work for us
+     * and TI does not supply a specific function.
+     * See ti81xx_nand.c::ti81xx_read_page_bch(), the e.g. offset calculations
+     * are all different. So we don't use a bbt for now. 
+
 	if (!(chip->options & NAND_BBT_SCANNED)) {
 		chip->options |= NAND_BBT_SCANNED;
 		chip->scan_bbt(mtd);
 	}
 
 	if (!chip->bbt)
+        */
 		return chip->block_bad(mtd, ofs, getchip);
 
-	/* Return info from the table */
-	return nand_isbad_bbt(mtd, ofs, allowbbt);
+	/* Return info from the table * /
+	return nand_isbad_bbt(mtd, ofs, allowbbt); */
 }
 
 /*
