diff --git a/include/configs/ti8148_evm.h b/include/configs/ti8148_evm.h
index a4539da..fb3c127 100644
--- a/include/configs/ti8148_evm.h
+++ b/include/configs/ti8148_evm.h
@@ -23,7 +23,7 @@
 #define CONFIG_TI814X_EVM_DDR3
 
 #define CONFIG_ENV_SIZE			0x800
-# define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + (8 * 1024))
+# define CONFIG_SYS_MALLOC_LEN		(CONFIG_ENV_SIZE + (196 * 1024))
 #define CONFIG_SYS_GBL_DATA_SIZE	128	/* size in bytes reserved for initial data */
 
 #ifndef TI81XX_SECOND_STAGE

