From 066df65dd436cd0e00fae480cf0966b01cd1ea91 Mon Sep 17 00:00:00 2001
From: Eik Binschek <binschek@dresearch-fe.de>
Date: Wed, 14 Jun 2012 10:13:32 +0200
Subject: [PATCH] u-boot: bootconfig print format enhancement

	change print output of u-boot

Signed-off-by: Eik Binschek <binschek@dresearch-fe.de>
---

diff --git a/common/main.c b/common/main.c
index abdb0ab..c3e83fc 100644
--- a/common/main.c
+++ b/common/main.c
@@ -388,10 +388,10 @@ void main_loop (void)
 	{
 		printf ("NAND-BootConfig:\n");
 	        printf ("Epoch = %d\n", btblock_test->epoch);
-	        printf ("Kernel booted = %d\n", btblock_test->kernel.n_booted );
-	        printf ("Kernel healthy = %d\n", btblock_test->kernel.n_healthy );
-	        printf ("Rootfs booted = %d\n", btblock_test->rootfs.n_booted );
-	        printf ("Rootfs healthy = %d\n", btblock_test->rootfs.n_healthy );
+	        printf ("Kernel not booted  = %d\n", btblock_test->kernel.n_booted );
+	        printf ("Kernel not healthy = %d\n", btblock_test->kernel.n_healthy );
+	        printf ("Rootfs not booted  = %d\n", btblock_test->rootfs.n_booted );
+	        printf ("Rootfs not healthy = %d\n", btblock_test->rootfs.n_healthy );
 
 	        /* rootfs_device=/dev/blockrom4\0       */
 		/* kernel_offset=0x200000\0 */
@@ -428,10 +428,10 @@ void main_loop (void)
 	                bc_ll_set_booted_flag_on_current( &bc );
 	        	printf ("NEW NAND-BootConfig:\n");
 		        printf ("Epoch = %d\n", btblock_test->epoch);
-	        	printf ("Kernel booted = %d\n", btblock_test->kernel.n_booted );
-	        	printf ("Kernel healthy = %d\n", btblock_test->kernel.n_healthy );
-	        	printf ("Rootfs booted = %d\n", btblock_test->rootfs.n_booted );
-	        	printf ("Rootfs healthy = %d\n", btblock_test->rootfs.n_healthy );
+	        	printf ("Kernel not booted  = %d\n", btblock_test->kernel.n_booted );
+	        	printf ("Kernel not healthy = %d\n", btblock_test->kernel.n_healthy );
+	        	printf ("Rootfs not booted  = %d\n", btblock_test->rootfs.n_booted );
+	        	printf ("Rootfs not healthy = %d\n", btblock_test->rootfs.n_healthy );
 	        }
 	}

-- 
1.7.7

